### PR TITLE
Media with content block #140

### LINKED ISSUE #140: Media with Content block
**Agile Requirement**
As a business owner, I want to be able to display an media with some text that can be followed by icons so that I can display important information about tecnical details of the engines.  

**Description** 
Create a Media with Content block so that an editor can add an image or video, text and 2 icons (optionally) with text as in the[ figma design. ](https://www.figma.com/file/URJhT8pkntqc6h6v5wSLqx/Mack-Core-Design-Kit?type=design&node-id=478-71521&mode=design&t=iL9idKrdD26rsMej-4)

This variant is displayed in the [solutions (secondary pdp) page ](https://www.figma.com/file/oNwe2jHIICPMIe551EY62l/Mack-Reskin---Full-Project?type=design&node-id=2243-137856&mode=dev).

**Acceptance Criteria**
1- As an editor, I can add the new block to any page. 
2- As an editor, If I put the image first in the word document it will be displayed to the left and if it's positioned second, then it will be displayed to the right. 
3- As an editor, to configure this block I can add a video or an image, a header and some descriptive text. Optionally I can also use up 2 images as "icons" to imitate the figma design. 
4- As an editor, it is my responsibility to add 2 icons in case I add any and the text related to these images is part of the image. 
5- As a user, when I click on a video, the video modal is opened and the video is reproduced in it.

### PR DESCRIPTION
**Pull request by SantiagoHomps-NC**
Fix #140 

Test URLs:
- Before: https://main--vg-macktrucks-com-rd--netcentric.hlx.page/drafts/shomps/redesign/media-with-content
- After: https://140-media-content--vg-macktrucks-com-rd--netcentric.hlx.page/drafts/shomps/redesign/media-with-content

### CODE REVIEW

**Review by jonatan-lledo-netcentric — APPROVED** [review]

File: blocks/v2-media-with-content/v2-media-with-content.css [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+.v2-media-with-content__text {
+  font-size: var(--body-font-size-s);
+  margin: 24px 0 0;
+}
+
+.v2-media-with-content__icon-list {
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  column-gap: 23px;
```
Comment (cogniSyb) [inline blocks/v2-media-with-content/v2-media-with-content.css]: Should be rather 24px

File: blocks/v2-media-with-content/v2-media-with-content.css [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+.redesign-v2 .section .v2-media-with-content-wrapper {
+  padding: 0;
+}
+
+.v2-media-with-content__content-wrapper {
+  display: flex;
+}
+
+.v2-media-with-content__content-wrapper.image-left {
```
Comment (cogniSyb) [inline blocks/v2-media-with-content/v2-media-with-content.css]: Default state shouldn’t use a 'modifier' class. Remove `.image-left` from the selector

File: blocks/v2-media-with-content/v2-media-with-content.css [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  .section.v2-media-with-content-container {
+    padding: 80px 0;
+  }
+
+  .redesign-v2 .section .v2-media-with-content-wrapper {
+    max-width: unset;
+  }
+
+  .v2-media-with-content__content-wrapper.image-right,
+  .v2-media-with-content__content-wrapper.image-left {
```
Comment (cogniSyb) [inline blocks/v2-media-with-content/v2-media-with-content.css]: Could this selector be simplified to `.v2-media-with-content__content-wrapper`?

File: blocks/v2-media-with-content/v2-media-with-content.css [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+.v2-media-with-content__content-wrapper {
+  display: flex;
+}
+
+.v2-media-with-content__content-wrapper.image-left {
+  flex-direction: column;
+}
+
+.v2-media-with-content__content-wrapper.image-right {
```
Comment (cogniSyb) [inline blocks/v2-media-with-content/v2-media-with-content.css]: BEM should be used, rewrite to `.v2-media-with-content__content-wrapper--image-right`

File: blocks/v2-media-with-content/v2-media-with-content.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    } else {
+      col.classList.add('column-with-text');
+    }
+  });
+
+  const hasImageLeft = (content[0].firstElementChild.tagName === 'PICTURE');
+  if (hasImageLeft) {
+    contentWrapper.classList.add('image-left');
+  } else {
+    contentWrapper.classList.add('image-right');
```
Comment (cogniSyb) [inline blocks/v2-media-with-content/v2-media-with-content.js]: `${blockName}__content-wrapper--image-right`

File: blocks/v2-media-with-content/v2-media-with-content.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  content.forEach((col) => {
+    col.classList.add(`${blockName}__column`);
+    if (col.firstElementChild.tagName === 'PICTURE') {
+      col.classList.add('column-with-image');
+    } else {
+      col.classList.add('column-with-text');
+    }
+  });
+
+  const hasImageLeft = (content[0].firstElementChild.tagName === 'PICTURE');
```
Comment (cogniSyb) [inline blocks/v2-media-with-content/v2-media-with-content.js]: The image-left class is not necessary. This const could be removed, the following code to be changed to a ternary statement

**Review by cogniSyb — CHANGES_REQUESTED** [review]

File: blocks/v2-media-with-content/v2-media-with-content.css [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+  flex-direction: column-reverse;
+}
+
+.v2-media-with-content__column.column-with-text {
+  padding: 24px 16px 0;
+}
+
+.v2-media-with-content__column img {
+  display: block;
+  width: 100vw;
```
Comment (cogniSyb) [inline blocks/v2-media-with-content/v2-media-with-content.css:24]: Add a max-width so that this image:


doesn’t overlap 1440px as seen below:

**Review by cogniSyb — APPROVED** [review]

**Review by jonatan-lledo-netcentric — APPROVED** [review]

**Pull request merged**